eox_nelp/programs/api/v1/views.py

```diff
@@ -34,6 +46,32 @@ def wrapper(self, request, *args, **kwargs):
     return decorator
 
 
+def process_national_id_query_param():
+    """Decorator to check if national_id query parameter is valid if provided.
+    If not provided, the view will proceed without filtering by national_id.
+    Using the request user if not provided. Added to the request object the nid_user attribute.
+    Returns  422 if invalid.
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, request, *args, **kwargs):
+            if national_id := self.request.query_params.get("national_id"):
+                if not is_valid_national_id(national_id):
+                    return Response(
+                        {
+                            "error": "INVALID_NATIONAL_ID",
+                            "message": "national_id must be digits only and 10–15 characters."
+                        },
+                        status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    )
+                setattr(request, 'nid_user', get_object_or_404(User, extrainfo__national_id=national_id))
```

Comment:
for me `nid_user` means `nationa_id_user` so I'm expecting  something like 123456789 from `request.nid_user` but this returns a user, so,could you change the name to something  clearer